### PR TITLE
fix(core): revisit dependencies w/ possibly higher distance

### PR DESCRIPTION
**Pull request by amadeusjposchmann**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [X] The commit message follows our guidelines: https://github.com/nestjs/nest/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [X] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: #11672 

## What is the new behavior?
Module references are now evaluated multiple times, accommodating for distance updates by other dependency chains containing the same reference.

## Does this PR introduce a breaking change?
- [ ] Yes
- [X] No

## Other information

### CODE REVIEW

**Comment by amadeusjposchmann** [comment]
Note: I didn't see any existing tests for correctness of module distances, but maybe I missed them?

**Comment by amadeusjposchmann** [comment]
Huh, strange. I didn't touch the file with the reduced coverage. Did I miss something?

File: packages/core/scanner.ts [hunk 1]
```diff
@@ -357,7 +357,7 @@ export class DependenciesScanner {
 
     const modulesStack = [];
     const calculateDistance = (moduleRef: Module, distance = 1) => {
-      if (!moduleRef || modulesStack.includes(moduleRef)) {
+      if (!moduleRef) {
```
Comment (kamilmysliwiec) [inline packages/core/scanner.ts:360]: I believe this may lead to infinite loops for circular improts

Comment (kamilmysliwiec) [reply packages/core/scanner.ts:360]: > True, but circular imports can't be allowed anyways, can they?

They are allowed & supported (even though it's against best practices, as stated in the docs).

Comment (amadeusjposchmann) [reply packages/core/scanner.ts:360]: The please re-open #11672. It gives an example of a non-circular dependency not being solved correctly and I don't see a way to make this work while supporting circular dependencies at the same time.

**Pull request closed without merging**